fiscalyear.py

```diff
@@ -603,6 +607,218 @@ def __ge__(self, other):
+
+    __slots__ = ['_fiscal_year', '_fiscal_month']
+
+    def __new__(cls, fiscal_year, fiscal_month):
+        """Constructor.
+
+        :param fiscal_year: The fiscal year
+        :type fiscal_year: int or str
+        :returns: A newly constructed FiscalMonth object
+        :rtype: FiscalMonth
+        :raises TypeError: If fiscal_year or fiscal_month is not
+            an int or int-like string
+        :raises ValueError: If fiscal_year or fiscal_month is out of range
+        """
+        fiscal_year = _check_year(fiscal_year)
+        fiscal_month = _check_month(fiscal_month)
+
+        self = super(FiscalMonth, cls).__new__(cls)
+        self._fiscal_year = fiscal_year
+        self._fiscal_month = fiscal_month
+        return self
+
+    @classmethod
+    def current(cls):
+        """Alternative constructor. Returns the current FiscalMonth.
+
+        :returns: A newly constructed FiscalMonth object
+        :rtype: FiscalMonth
+        """
+        today = FiscalDate.today()
+        return cls(today.fiscal_year, today.fiscal_month)
+
+    def __repr__(self):
+        """Convert to formal string, for repr().
+
+        >>> fm = FiscalMonth(2017, 1)
+        >>> repr(fm)
+        'FiscalMonth(2017, 1)'
+        """
+        return '%s(%d, %d)' % (self.__class__.__name__,
+                               self._fiscal_year,
+                               self._fiscal_month)
+
+    def __str__(self):
+        """Convert to informal string, for str().
+
+        >>> fm = FiscalMonth(2017, 1)
+        >>> str(fm)
+        'FY2017 FM1'
+        """
+        return 'FY%d FM%d' % (self._fiscal_year, self._fiscal_month)
+
+    # TODO: Implement __format__ so that you can print
+    # fiscal year as 17 or 2017 (%y or %Y)
+
+    def __contains__(self, item):
+        """Returns True if item in self, else False.
+
+        :param item: The item to check
+        :type item: FiscalMonth, FiscalDateTime,
+            datetime, FiscalDate, or date
+        :rtype: bool
+        """
+        if isinstance(item, FiscalMonth):
+            return self == item
+        elif isinstance(item, datetime.datetime):
+            return self.start <= item <= self.end
+        elif isinstance(item, datetime.date):
+            return self.start.date() <= item <= self.end.date()
+        else:
+            raise TypeError("can't compare '%s' to '%s'" % (
+                type(self).__name__, type(item).__name__))
+
+    # Read-only field accessors
+
+    @property
+    def fiscal_year(self):
+        """:returns: The fiscal year
+        :rtype: int
+        """
+        return self._fiscal_year
+
+    @property
+    def fiscal_month(self):
+        """:returns: The fiscal month
+        :rtype: int
+        """
+        return self._fiscal_month
+
+    @property
+    def start(self):
+        """:returns: Start of the fiscal month
+        :rtype: FiscalDateTime
+        """
+
+        calendar_month = ((self._fiscal_month - START_MONTH) % 12 + 6) % 12 + 1
```

Comment:
The logic on this line is a bit confusing to me. I think it would make more sense as:
```suggestion
        calendar_month = (START_MONTH + self._fiscal_month - 1) % 12
        if calendar_month == 0:
            calendar_month = 12
```
Does that look correct?

Reply:
Yeah that definitely looks better. Passes the tests too. Admittedly I struggled with this conversion and just ended up brute forcing it until something worked. I may also have gotten hung up trying to make it a one-liner, but it's been a while and I honestly don't remember.

Reply:
Pushed up the change